integrations/extensions/docs/elasticsearch-install-and-setup/how_to_use_web_crawler_in_elasticsearch.md

```diff
@@ -0,0 +1,284 @@
+
+
+* Click on `edit pipline` and then `Manage` -> `Edit`, it will take you to the ingest pipeline `Edit` page 
+where you can add processors to the pipeline.
+
+
+* Add a `Script` processor for chunking  
+  In the ingest pipeline page, click on `Add a processor`, choose `Script` processor, and then add a Groovy script to the `Source` field.  
+  For example,  
+  <img src="assets/web_crawler_script_processor.png" width="577" height="718" />
+
+  ```Groovy
+  String[] envSplit = /((?<!M(r|s|rs)\.)(?<=\.) |(?<=\!) |(?<=\?) )/.split(ctx['body_content']);
+  ctx['passages'] = new ArrayList();
+  int i = 0;
+  boolean remaining = true;
+  if (envSplit.length == 0) {
+      return
+  } else if (envSplit.length == 1) {
+      Map passage = ['text': envSplit[0]];ctx['passages'].add(passage)
+  } else {
+      while (remaining) {
+          Map passage = ['text': envSplit[i++]]
+          while (i < envSplit.length && passage.text.length() + envSplit[i].length() < params.model_limit) {passage.text = passage.text + ' ' + envSplit[i++]}
+          if (i == envSplit.length) {remaining = false}
+          ctx['passages'].add(passage)
+          }
+  }
+  ```
+  The above script splits `body_content` into sentences using regex, and then combine sentences into passages with 
+  a `model_limit` parameter to control the character length of each passage. `model_limit` is configured in `Parameters` field, for example,  
+  ```json
+  {
+    "model_limit": 2048
+  }
+  ```
+  
+  You can update the `Script` to build a customized chunking processor for your web content.  
+
+  For example, you can include `title` in each passage object with the following `passage` definition:
+  ```Groovy
+  Map passage = ['text': envSplit[i++], 'title': ctx['title']]
+  ```
+  If you want to insert `title` to the beginning of each chunked text,
+  ```Groovy
+  Map passage = ['text': ctx['title'] + '.' + envSplit[i++], 'title': ctx['title']]
+  ```
+
+
+* Add a `Foreach` processor to process chunked texts using the ELSER model  
+  In the ingest pipeline `Edit` page, click on `Add a processor`, choose `Foreach` processor, and then add a JSON processor config to the `Processors` field.  
+  For example,  
+  <img src="assets/web_crawler_foreach_processor.png" width="575" height="716" />
+  ```json
+  {
+    "inference": {
+      "field_map": {
+        "_ingest._value.text": "text_field"
+      },
+      "model_id": ".elser_model_1",
+      "target_field": "_ingest._value.sparse",
+      "inference_config": {
+        "text_expansion": {
+          "results_field": "tokens"
+        }
+      },
+      "on_failure": [
+        {
+          "append": {
+            "field": "_source._ingest.inference_errors",
+            "value": [
+              {
+                "message": "Processor 'inference' in pipeline 'search-crawler-with-chunking@custom' failed with message '{{ _ingest.on_failure_message }}'",
+                "pipeline": "search-crawler-with-chunking@custom",
```

Comment:
The name `search-crawler-with-chunking@custom` will differ based on how the user names their pipeline. Should be explicitly called out to have the user update it after copy + paste